tools/automation.py

```diff
@@ -0,0 +1,316 @@
+    """
+    multiline_str = ""
+    if not pd.isna(long_text_arg):
+        multiline_str = LiteralScalarString(textwrap.dedent(long_text_arg))
+    return multiline_str
+
+
+def get_mentorship_type(mentorship_type_str):
+    """
+    Returns ad-hoc, long-term, both or empty str
+    """
+    mentorship_type = mentorship_type_str.lower()
+
+    if TYPE_AD_HOC in mentorship_type:
+        return TYPE_AD_HOC.replace(' ', '-')
+    elif TYPE_LONG_TERM in mentorship_type:
+        return TYPE_LONG_TERM
+    elif TYPE_BOTH in mentorship_type:
+        return TYPE_BOTH
+
+    return ""
+
+
+def update_yml_file_formatting(s):
+    """
+    Insert new line before each new mentor.
+    Replace yaml string formatting indicator '|-' with '|'
+    """
+    updated_string = s.replace('- name: ', '\n- name: ').replace('|-', '|')
+    return updated_string
+
+
+def write_yml_file(file_path, mentors_data, mode):
+    """
+    Create new or append to mentors.yml file
+    :mentors_data: list of dictionaries
+    :mode: write or append
+    """
+    with open(file_path, mode, encoding = "utf-8") as output_yml:
+        yaml = YAML()
+
+        # Display yaml content in block style (not inline)
+        yaml.default_flow_style = False
+
+        # TODO: Currently do not use indent() - discuss with Adriana
+        # yaml.indent(mapping=2, sequence=4, offset=2)
+
+        yaml.dump(mentors_data, output_yml, transform=update_yml_file_formatting)
+    print(f"File: {file_path} is successfully written.")
+
+
+def read_yml_file(file_path):
+    """
+    Read yml file
+    """
+    with open(file_path, 'r', encoding="utf-8") as input_yml:
+        yaml=YAML(typ='safe')
+        yml_dict = yaml.load(input_yml)
+        print(f"File: {file_path} is successfully read.")
+    return yml_dict
+
+
+def xlsx_to_yaml_parser(mentor_row, mentor_index):
+    """
+    Prepare mentor's excel data for yaml format
+    """
+    mentee_str = get_multiline_string(mentor_row.iloc[28])
+    bio_str = get_multiline_string(mentor_row.iloc[10])
+    extra_str = get_multiline_string(mentor_row.iloc[11])
+
+    areas = get_yaml_block_sequence(mentor_row,
+                                    AREAS_START_INDEX,
+                                    AREAS_END_INDEX)
+    focus = get_yaml_block_sequence(mentor_row,
+                                    FOCUS_START_INDEX,
+                                    FOCUS_END_INDEX)
+    programming_languages = get_yaml_block_sequence(mentor_row,
+                                                    PROG_LANG_START_INDEX,
+                                                    PROG_LANG_END_INDEX)
+
+    type_of_mentorship = get_mentorship_type(mentor_row.iloc[3])
+    network_list = get_social_media_links(mentor_row.iloc[30], mentor_row.iloc[31])
+    hours_per_month = extract_numbers_from_string(mentor_row.iloc[29])
+    max_experience = extract_numbers_from_string(mentor_row.iloc[9])
+
+    # TODO: After testing phase change to mentor_disabled to True
```

Comment:
I don't think we need that, we will review via PR. It should be fine the default be False.

Reply:
Ok